database/database.go

```diff
@@ -178,6 +171,19 @@ func (db *Database) Protocols() []p2p.Protocol {
 	return nil
 }
 
+func (db *Database) UpdateExecuteTxMetrics(tx *ethTypes.Transaction) {
```

Comment:
[MINOR] I cannot see reason to encapsulate this code on a separate method when there is only one usage of it in the whole code. This encapsulate brings more complexity and not reusability. I would rather mode this code back to `ExecuteTx` unless you have already an usage in mind for having it separated

Reply:
The reason for encapsulating the metrics code is that it's not suppose to be in the important method such as ExecuteTx and poluting it in first place but should have been done inside the metrics package. I also encapsulated it also to keep the SRP of ExecuteTx as it was doing more stuff than just metrics